common-libs/common/src/services/DidDocumentService/elem-lib/op.ts

```diff
@@ -18,20 +19,20 @@ export const getDidDocumentModel = (primaryPublicKey: string, recoveryPublicKey:
   ],
 })
 
-export const getCreatePayload = (didDocumentModel: any, primaryKey: any) => {
+export const getCreatePayload = (didDocumentModel: any, signFn: (payload: Buffer) => Buffer) => {
   // Create the encoded protected header.
   const header = {
     operation: 'create',
     kid: '#primary',
     alg: 'ES256K',
   }
-  return makeSignedOperation(header, didDocumentModel, primaryKey.privateKey)
+  return makeSignedOperation(header, didDocumentModel, signFn)
 }
 
-export const makeSignedOperation = (header: any, payload: any, privateKey: any) => {
+export const makeSignedOperation = (header: any, payload: any, signFn: (payload: Buffer) => Buffer) => {
   const encodedHeader = encodeJson(header)
   const encodedPayload = encodeJson(payload)
-  const signature = signEncodedPayload(encodedHeader, encodedPayload, privateKey)
+  const signature = base64url.encode(signEncodedPayload(encodedHeader, encodedPayload, signFn))
```

Comment:
The original version of the code called `base64url.encode` on `signatureObject.signature`. The new one calls it on `Buffer.from(signatureObject.signature)`. Not sure what is the point of this change, and not sure if it doesn't break backwards compatibility?

Reply:
The type of parameters changed from `string` to `Buffer` to avoid some unnecessary encoding/decoding of hex string to buffer.

The tests are still passing, so backward compatibility is still there. Are there other tests that are missing that need to be added?

Reply:
But in this case it added unnecessary decoding. The original version executed `base64url.encode` directly on `signatureObject.signature`, while the new one additionally calls `Buffer.from` (in keyvault) prior to `base64url.encode`.

> The tests are still passing, so backward compatibility is still there.

Generally tests only check that the current version correctly decodes what it encoded etc.

Do we have some special tests in this case that would check that the old version correctly decodes what the new version has encoded and vice versa?

Reply:
https://github.com/affinityproject/affinidi-core-sdk/blob/master/common-libs/common/test/unit/services/DidDocumentService.test.ts#L121 Explicitly compares the output with a previously generated did, is it not sufficient?

I'll change the `sign` method to return `Uint8Array` which makes more sense than Buffer, we don't expect to edit the signature.

Reply:
The change to `Uint8Array` still requires `Buffer.from` in other parts of the code to transform the buffer to hex, let's leave it as is for now.